polars/dataframe.ts

```diff
@@ -3130,6 +3172,16 @@ function DataFrameConstructor<S extends Schema = Schema>(
   data?,
   options?,
 ): DataFrame<S> {
+  if (data?.schema && !Array.isArray(data?.schema)) {
+    return _DataFrame(
+      new pli.JsDataFrame(
+        Object.keys(data.schema).map((key) =>
+          Series(key, [], data.schema[key]).inner(),
+        ),
+      ),
+    );
+  }
+
```

Comment:
So this will error if theres an object with the name of schema and its a series

```ts
pl.DataFrame({
  "schema": pl.Series([1,2,3]),
})
```

it also fails if its an array buffer like Float64Array
```ts
pl.DataFrame({
  "schema": Float64Array.of(1,2,3)
})
```